validator/client/runner.go

```diff
@@ -128,6 +147,8 @@ func run(ctx context.Context, v iface.Validator) {
 				continue
 			}
 			performRoles(slotCtx, allRoles, v, slot, &wg, span)
+		case currentKeys := <-accountsChangedChan: // should be less of a priority than next slot
```

Comment:
events should take place before nextslot right? the event may do something that changes the duties in the slot

Reply:
For that to happen the event would need to take place right at the start of the slot. The only event that can cause this is a reorg, but I don't think it happens at slot time 0. On the downside, if for whatever reason there is a bunch of events to process or an event takes long to process, it will delay other more important things.

Reply:
ok i implemented something that should work well for this case